File: Directory/2. Add Two Numbers/Solution.cpp
class Solution {
public:
    ListNode* addTwoNumbers(ListNode* l1, ListNode* l2) {
        ListNode *ret = new ListNode(0);
        ListNode *cur = ret;
        int sum = 0;
        while (true) {
            if (l1 != NULL) {
                sum += l1->val;
                l1 = l1->next;
            }
            if (l2 != NULL) {
                sum += l2->val;
                l2 = l2->next;
            }
            cur->val = sum % 10;
            sum /= 10;
            if (l1 != NULL || l2 != NULL || sum)
                cur = (cur->next = new ListNode(0));
            else
                break;
        }
        return ret;
    }
};

File: Directory/2. Add Two Numbers/Solution.py
# -*- coding: utf-8 -*-
# @Author: yufeng
# @Date:   2018-10-13 19:27:56
# @Last Modified by:   yufeng
# @Last Modified time: 2018-10-13 19:56:07
# @Email: fzhiy270@sina.com
# @Blog: http://fzhiy.com/


# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def addTwoNumbers(self, l1, l2):
        """
        :type l1: ListNode
        :type l2: ListNode
        :rtype: ListNode
        """
        res = cur = ListNode(0)
        carry = 0
        while l1 or l2 or carry:
        	if l1:
        		carry, l1 = carry+l1.val, l1.next
        	if l2:
        		carry, l2 = carry+l2.val, l2.next
        	cur.next = ListNode(carry%10)
        	cur = cur.next
        	carry //= 10
        return res.next # res会有前导0
        